Tests/test_Seq_objs.py

```diff
@@ -737,6 +737,184 @@ def test_init_typeerror(self):
+
+    def test_join_Seq_with_file(self):
+        """Checks if Seq join correctly concatenates sequence from a file with the spacer."""
+        filename = 'Fasta/f003'
+        seqlist = [record.seq for record in SeqIO.parse(filename, 'fasta')]
+        seqlist_as_strings = [str(_) for _ in seqlist]
+
+        spacer = Seq('NNNNN')
+        spacer1 = Seq('')
+        # seq objects with spacer
+        seq_concatenated = spacer.join(seqlist)
+        # seq objects with empty spacer
+        seq_concatenated1 = spacer1.join(seqlist)
+
+        ref_data = ref_data1 = ""
+        ref_data = str(spacer).join(seqlist_as_strings)
+        ref_data1 = str(spacer1).join(seqlist_as_strings)
+
+        self.assertEqual(str(seq_concatenated), ref_data)
+        self.assertEqual(str(seq_concatenated1), ref_data1)
+        self.assertEqual(NotImplemented, spacer.join(SeqIO.parse(filename, 'fasta')))
+
+    def test_join_UnknownSeq(self):
+        """Checks if UnknownSeq join correctly concatenates sequence with the spacer."""
+        # Only expect it to take Seq objects and/or strings in an iterable!
+
+        spacer1 = UnknownSeq(0, character="-", alphabet=generic_dna)
+        spacers = [spacer1, UnknownSeq(5, character="-", alphabet=generic_dna), UnknownSeq(5, character="-", alphabet=generic_nucleotide)]
+
+        example_strings = ["ATG", "ATG", "ATG", "ATG"]
+        example_strings_seqs = ["ATG", "ATG", Seq("ATG", generic_dna), "ATG"]
+
+        # strings with empty spacer
+        str_concatenated = spacer1.join(example_strings)
+
+        self.assertEqual(str(str_concatenated), "".join(example_strings))
+        self.assertEqual(str_concatenated.alphabet, spacer1.alphabet)
+
+        for spacer in spacers:
+            seq_concatenated = spacer.join(example_strings_seqs)
+            self.assertEqual(str(seq_concatenated), str(spacer).join(example_strings))
+            self.assertEqual(seq_concatenated.alphabet, spacer.alphabet)
+
+    def test_join_UnknownSeq_with_file(self):
+        """Checks if UnknownSeq join correctly concatenates sequence from a file with the spacer."""
+        filename = 'Fasta/f003'
+        seqlist = [record.seq for record in SeqIO.parse(filename, 'fasta')]
+        seqlist_as_strings = [str(_) for _ in seqlist]
+
+        spacer = UnknownSeq(0, character="-", alphabet=generic_dna)
+        spacer1 = UnknownSeq(5, character="-", alphabet=generic_dna)
+        # seq objects with spacer
+        seq_concatenated = spacer.join(seqlist)
+        # seq objects with empty spacer
+        seq_concatenated1 = spacer1.join(seqlist)
+
+        ref_data = ref_data1 = ""
+        ref_data = str(spacer).join(seqlist_as_strings)
+        ref_data1 = str(spacer1).join(seqlist_as_strings)
+
+        self.assertEqual(str(seq_concatenated), ref_data)
+        self.assertEqual(str(seq_concatenated1), ref_data1)
+        self.assertEqual(NotImplemented, spacer.join(SeqIO.parse(filename, 'fasta')))
```

Comment:
See comment on the implementation, this should probably be ``self.assertRaises(...)`` instead.